internal/typeinfo/slice_test.go

```diff
@@ -0,0 +1,45 @@
+package typeinfo
+
+import (
+	"reflect"
+
+	. "gopkg.in/check.v1"
+)
+
+func (*typeInfoSuite) TestLocateParamsSlice(c *C) {
+	type S []any
+
+	argInfo, err := GenerateArgInfo([]any{S{}})
+	c.Assert(err, IsNil)
+
+	s := S{1, 2}
+	valOfS := reflect.ValueOf(s)
+	typeToValue := map[reflect.Type]reflect.Value{
+		reflect.TypeOf(s): valOfS,
+	}
+
+	input, err := argInfo.InputSlice("S")
+	c.Assert(err, IsNil)
+
+	vals, err := input.LocateParams(typeToValue)
+	c.Assert(err, IsNil)
+	c.Assert(vals, HasLen, len(s))
+
+	for i := 0; i < len(s); i++ {
+		c.Assert(vals[i].Interface(), Equals, s[i])
+	}
+}
+
+func (*typeInfoSuite) TestLocateParamsSliceError(c *C) {
```

Comment:
If adding table tests above this can probably be added as a test case.

Reply:
I would prefer to leave this as it is. Generally in the codebase we are keeping the error tests separate from the acceptance tests. I did try this as a table driven test in its own right but this case doesn't really suit it very well.

Reply:
Both ways seem to be standard in Go, I agree with you that it is the convention to separate them in SQLair.

Reply:
And even in this case I can think of a couple of more tests like having a known type and an unknown one, only unknown, etc.

Reply:
I've added that extra tests but its really just testing a map lookup and error message that is at the start of all `LocateParams` methods rather than any functionality related to slices. I don't think there's anything else to really add here, the sanitizing of the inputs is done by `bindinputs` and has already been handled by the time `LocateParams` is called.